pkg/webhooks/webhookpatch.go

```diff
@@ -97,34 +89,51 @@ func PatchCertLoop(injectionWebhookConfigName, webhookName, caBundlePath string,
 				if oldConfig.ResourceVersion != newConfig.ResourceVersion {
 					for i, w := range newConfig.Webhooks {
 						if w.Name == webhookName && !bytes.Equal(newConfig.Webhooks[i].ClientConfig.CABundle, caCertPem) {
-							log.Infof("Detected a change in CABundle, patching MutatingWebhookConfiguration again")
-							shouldPatch <- struct{}{}
+							log.Infof("Detected a change in CA bundle, patching MutatingWebhookConfiguration for %s", newConfig.Name)
+							// either this is the canonical webhook for the revision and we should keep trying to patch
+							// the CABundle forever or it is a revision tag webhook and we should try once since blindly
+							// retrying could lead to overwriting a webhook that's no longer labeled with our revision
```

Comment:
If I understand right, this is to ensure if another revision takes over we don't overwrite it?

if this happens between the time we get the UpdateEvent and our write, it will fail with a conflict. At that point, we can do a Get on the webhook and ensure it still has the label and bail out?

The benefit I think would be to simplify this so that there is one path, and make it so we can retry the tag webhooks to account for temporary api-server downtime

Reply:
Makes sense, changed it so that there's one path that checks for the revision and bails out of the retry if it's not found or doesn't match. Completely replaced the old retry logic for this so it's worth a closer look